utils/python/CIME/case_setup.py

```diff
@@ -190,6 +190,7 @@ def _case_setup_impl(case, caseroot, clean=False, test_mode=False, reset=False):
                    "it is not possible to run with OpenMP if using the NAG Fortran compiler")
             cost_pes = env_mach_pes.get_cost_pes(pestot, thread_count, machine=case.get_value("MACH"))
             case.set_value("COST_PES", cost_pes)
+            case.set_value("TOTAL_CORES", pestot * thread_count)
```

Comment:
@jedwards4b melvin's current config is for PES_PER_NODE=64 and MAX_TASKS_PER_NODE=64, the behavior I'm seeing for COST_PES is still rounding up to 64. I'm wondering how to support your SMT case and the basic desktop case of 1-thread per core.